File: instruct_generations/cpp/files/question-423-pass-1.cpp
#include <bits/stdc++.h> 
 using namespace std ;
 bool modularSum ( int arr [ ] , int n , int m ) {
// Initialize result
    bool result = false;

    // Iterate through the array
    for (int i = 0; i < n; i++)
    {
        // Initialize sum
        int sum = 0;

        // Iterate through the remaining elements of the array
        for (int j = i; j < n; j++)
        {
            // Add the element to the sum
            sum += arr[j];

            // Check if the sum is divisible by m
            if (sum % m == 0)
            {
                // If it is, return true
                result = true;
                break;
            }
        }

        // If the sum is divisible by m, break out of the loop
        if (result)
            break;
    }

    // Return the result
    return result;
}

# Subset with sum divisible by m.
#include <bits/stdc++.h> 
using namespace std ;
bool modularSum ( int arr [ ], int n, int m ) {

File: fine_tuning/python_preprocessing/orig_files/question-423-pass-1.py
arr = [100, 1000, 100, 1000, 1]
sum = [-1] * 10000


def maxSumWO3Consec(n):
    if (sum[n] != -1):
        return sum[n]
    if (n == 0):
        sum[n] = 0
        return sum[n]
    if (n == 1):
        sum[n] = arr[0]
        return sum[n]
    if (n == 2):
        sum[n] = arr[1] + arr[0]
        return sum[n]
    sum[n] = max(max(maxSumWO3Consec(n - 1),
                     maxSumWO3Consec(n - 2) + arr[n]),
                 arr[n] + arr[n - 1] + maxSumWO3Consec(n - 3))
    return sum[n]


if __name__ == "__main__":
    n = len(arr)
    print(maxSumWO3Consec(n))
